bundles/azure/aso/psql/bundle/config/00-schema.yml

```diff
@@ -0,0 +1,143 @@
+#@data/values-schema
+---
+#@schema/title "ASO Controller Namespace"
+#@schema/desc "The Namespace where the Azure ASO controller is installed that should own this Azure RM resource"
+aso_controller_namespace: ""
+
+#@schema/title "ResourceName"
+#@schema/desc "Name for the resources"
+name: "aso-psql"
+
+#@schema/title "ResourceNamespace"
+#@schema/desc "Kubernetes namespace where the Azure resources will be created"
+namespace: ""
+
+#@schema/title "Create namespace flag"
+#@schema/desc "Whether to create or not the namespace for the resources"
+create_namespace: True
+
+#@schema/title "ResourceGroup"
+#@schema/desc "Azure ResourceGroup for the servers/database resources"
+resource_group:
+  #@schema/title "Name"
+  #@schema/desc "Azure ResourceGroup name"
+  name: ""
+  #@schema/title "UseExisting"
+  #@schema/desc "Whether to use the existing Azure resource group or not"
+  use_existing: False
+  #@schema/title "Tags"
+  #@schema/desc "Tags to attach to the object"
+  #@schema/default []
+  tags:
+    -
+      #@schema/title "Key"
+      #@schema/desc "The name of the tag"
+      #@schema/nullable
```

Comment:
_question_: why are `key` and `value` `nullable`? Would the resource group API allow for any of these be missing / null? 

(Same goes for other `tags` and their `keys`/`values`)

Reply:
In order to define a type for a field I have to set it to a value of that specific type. For example, I want the key to be a string, thus I set "thekey" as value. This means that, if I don't set a value for it in my values file, it gets set to whatever value has been set in the schema, which is "thekey" in our case. Adding the `schema/nullable` annotation makes it possible for it to assume a _null_ value, which I can validate later on.
I found no way to make it a mandatory field without assuming the value defined in the schema. Can you advise a better solution?

Reply:
> This means that, if I don't set a value for it in my values file, it gets set to whatever value has been set in the schema, which is "thekey" in our case.

No, the value for `tags` would be `[]`, because you set `#@schema/default []` Only if you set only parts of the inner map (e.g. only a `key` but no `value` it would default `value` to `"thevalue"`. Not knowing the APIs, I am not sure what's better, having the values default to `null` or `""`.

<details><summary>example</summary>

```console
$ grep -R ^ s
s/stuff.yaml:#@ load("@ytt:data", "data")
s/stuff.yaml:
s/stuff.yaml:---
s/stuff.yaml:stuff: #@ data.values.tags
s/values.yaml:#@data/values-schema
s/values.yaml:
s/values.yaml:---
s/values.yaml:#@schema/default []
s/values.yaml:tags:
s/values.yaml:-
s/values.yaml:  key: ""
s/values.yaml:  value: ""
$ ytt -f s
stuff: []
$ ytt -f s --data-values-schema-inspect -o openapi-v3
openapi: 3.0.0
info:
  version: 0.1.0
  title: Schema for data values, generated by ytt
paths: {}
components:
  schemas:
    dataValues:
      type: object
      additionalProperties: false
      properties:
        tags:
          type: array
          items:
            type: object
            additionalProperties: false
            properties:
              key:
                type: string
                default: ""
              value:
                type: string
                default: ""
          default: []
$ ytt -f s --data-value-yaml tags='[{"key":"some_key", "value": "some_value"}]'
stuff:
- key: some_key
  value: some_value
$ $ ytt -f s --data-value-yaml tags='[{"key":"some_key"}]'
stuff:
- key: some_key
  value: ""
$ ytt -f s --data-value-yaml tags='[{},{}]'
stuff:
- key: ""
  value: ""
- key: ""
  value: ""
$ 
```

</details>

Reply:
Are we missing a commit? I don't see any changes around that ... am I missing something?

Reply:
You're 100% right, I didn't commit the changes. Done now.
I'm also unresolving this conversation, please do resolve it when you think it's ok.